packages/grid/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -8,38 +8,27 @@ import {
   unstable_ownerDocument as ownerDocument,
   unstable_capitalize as capitalize,
 } from '@mui/utils';
-import { getDataGridUtilityClass } from '../../constants/gridClasses';
-import {
-  GridCellEventLookup,
-  GridEvents,
-  GridCellMode,
-  GridCellModes,
-  GridRowId,
-} from '../../models';
-import { GridAlignment } from '../../models/colDef/gridColDef';
+import { getDataGridUtilityClass, gridClasses } from '../../constants/gridClasses';
+import { GridCellEventLookup, GridEvents, GridCellModes, GridRowId } from '../../models';
+import { GridRenderEditCellParams, FocusElement } from '../../models/params/gridCellParams';
+import { GridColDef, GridAlignment } from '../../models/colDef/gridColDef';
+import { GridTreeNodeWithRender } from '../../models/gridRows';
+import { useGridSelector, shallowCompare } from '../../hooks/utils/useGridSelector';
 import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
 import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
 import { gridFocusCellSelector } from '../../hooks/features/focus/gridFocusStateSelector';
+import { gridEditRowsStateSelector } from '../../hooks/features/editing/gridEditingSelectors';
 import type { DataGridProcessedProps } from '../../models/props/DataGridProps';
-import { FocusElement } from '../../models/params/gridCellParams';
 
-export interface GridCellProps<V = any, F = V> {
+export interface GridCellProps {
   align: GridAlignment;
   className?: string;
   colIndex: number;
-  field: string;
+  column: GridColDef;
   rowId: GridRowId;
-  formattedValue?: F;
-  hasFocus?: boolean;
   height: number | 'auto';
-  isEditable?: boolean;
-  isSelected?: boolean;
   showRightBorder?: boolean;
-  value?: V;
   width: number;
-  cellMode?: GridCellMode;
-  children: React.ReactNode;
-  tabIndex: 0 | -1;
```

Comment:
These are technically breaking changes because these props are passed to the [`cell` slot](https://github.com/mui/mui-x/blob/f03017b40ac72095b342a84dc30ed833f10f62e7/packages/grid/x-data-grid/src/constants/defaultGridSlotsComponents.ts#L23).
We might break people's code in cases where they're wrapping the default `GridCell` component **and** accessing the props that have been removed:
```jsx
const CellSlot = (props) => {
  // props.value is not here anymore
  <GridCell {...props} />
}

<DataGrid
  slots={{
    cell: CellSlot
  }}
/>
```

I think it's a rare use case, but quickly searching through the issues, I found one example:
https://github.com/mui/mui-x/issues/8511#issuecomment-1524001995

Maybe we can postpone changes to this interface until the next major version?

Reply:
Damn, that's true. We lose quite a few benefits of this PR if we keep those fields where they are though. For example, if `hasFocus` is at the row level, then the whole row has to re-render when a cell gains/loses focus. If it's at the cell level, then only the specific cells that gained/lost focus need to re-render.

Do we have planned dates for major versions?

Reply:
What if instead of using `rootProps.slots.cell` in `GridRow`, we add a wrapper component (`GridCellWrapper`) that will render `rootProps.slots.cell` with the same props as before?
Moving the code from `GridRow` to `GridWrapperCell` instead of `GridCell` would allow keeping the same interface for `GridCell`.
Does this make sense?

Reply:
I've applied the wrapper thing, works great.